### PR TITLE
Simplify temporary_directory and use safe_rmtree

### PR DESCRIPTION
**Pull request by thejcannon**
Resolves: https://github.com/python-poetry/poetry/issues/3493

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

Unit test fails without changes and passes with changes 🎉

### CODE REVIEW

File: tests/utils/test_helpers.py [hunk 1]
```diff
@@ -62,3 +67,11 @@ def test_parse_requires():
 )
 def test_utils_helpers_canonical_names(raw):
     assert canonicalize_name(raw) == "a-b-c"
+
+
+def test_utils_helpers_temporary_directory_readonly_file():
```
Comment (finswimmer) [inline tests/utils/test_helpers.py:72]: Here a check whether the folder is removed successfully is missing, is it?

**Review by finswimmer — CHANGES_REQUESTED** [review]
Thanks a lot :+1: 

Just a minor thing.

File: tests/utils/test_helpers.py (hunk 1 above)
Comment (thejcannon) [reply tests/utils/test_helpers.py:72]: The bug itself would throw `PermissionError` on the removal attempt. So in that regard the test shows we no longer run into that issue.
That being said checking that it _actually_ was removed is probably also a good idea 😄

Comment (thejcannon) [reply tests/utils/test_helpers.py:72]: Alrighty, added some asserts. 👍

**Review by finswimmer — APPROVED** [review]
Thanks a lot and happy new year :rocket:

**Comment by Bettdatt** [comment]
Yes

**Pull request merged**